### ISSUE TITLE
TeamViewer 8 GUI elements are not accessible with NVDA when user join a meeting

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2944]
**Reported by janusz on 2013-01-25 11:28**
Dear developers,
I would like to valuably please You for one help.
TeamViewer 8 application is perfectly accessible till user join some meeting.
Do You think, that You could try to make some script for NVDA, which would activate The enable microphone and second script for disabling The microphone?
I Am aware, that MSAA objects tree is empty and that it will be very complex, but it is very unlikely possible, that TeamViewer 8 developers will make those objects accessible.
OCR function of NVDA can fetch text objects, but graphical objects of microphone enable or disable are not reachable even while using OCR addon of NVDA.
The onlyone object accessible by using NVDA navigation keys is TV0.
In teamviewer older version, version 6, one of TV0 objects would enable users to enable or disable The microphone by using right mouse click.

This application is giving NVDA and other screen readers developer A chance to start with new access to not accessible GUIS of applications.
Mr Tech, MR Curan. Does somebody of us know The power program built to The Windows Xp, Vista and later installations, Rundll32.exe
Does somebody tried to analyse some registry keys, that some apps are not only calling The procedures inside .dlls without arguments, but that even arguments can be passed to some calls by using comma?

What about triing to find The name of The procedure inside .exe program, and and recalling it in The right time?
For example, program will use some internal procedure when user activates The specific button.
If user.exe procedure for turning computer off can be used, eventhough user.exe resides in RAM and other programs and Windows dlls are communicating with it, what try to call also procedures of some inaccessible programs, if MSAA or display model of NVDA can not get necessary information?
The crucial problem is legality of this behaviour and licensing issues.
Developers will not be very glad to cooperate, because every body have his boss and in some cases, not every button is associated with standalone procedure name.
BUt like professionals. How to solve totally inaccessible GUIS, if there is no chance, that core developers of those programs will recompile and make necessary modifycation to The GUI of AN program?
Thank You very much for Yours cooperation and atleast for an professional discussion on this topic.
I donot know about other legal ways, than using MSAA, display model of NVDA or OCR.
One developer of Winamp special script used special API calls to control The process.
Please, try to think about my pleasment, atleast about algorithms for enabling and disabling The microphone.

I know, that my pleasments are very often not easy to implement, that there are many others, that can be implmenet much more easily.
Such as making special little .python script, which will make NVDA to automatically speak in some sys list wis, which CTRL+insert+F2 combination can speak, but when pressing arrow keys, NVDA can not speak those items automatically.
But i believe, that i will find some developers, who would try to atleast think about my pleasment and who would discuss with me.

**Comment by leonardder** [comment]
There is a Teamviewer add-on. I'm using TeamViewer 10 quite a lot, and you are able to access these features using object nav. TeamViewer is now at version 12, thus it is very unlikely that a fix will be provided for version 8. I suggest a close, and if still relevant, a new issue can be opened for the most recent version.
